### PR TITLE
Fix Non max suppression float16 kernel

### PR DESCRIPTION
**Pull request by fsx950223**
cc @mingxingtan

### CODE REVIEW

**Comment by mingxingtan** [comment]
I tested it for efficientdet and it worked, but looks like this PR changes existing API for v3 and v4, it might be difficult to check in this PR (due to the backward compatibility).

**Comment by fsx950223** [comment]
> I tested it for efficientdet and it worked, but looks like this PR changes existing API for v3 and v4, it might be difficult to check in this PR (due to the backward compatibility).

Yes, ```T_threshold``` leads to a bug when I test the test case that I added. I want to know is there a better way to keep iou_threshold and boxes the same type.

**Comment by fsx950223** [comment]
> I tested it for efficientdet and it worked, but looks like this PR changes existing API for v3 and v4, it might be difficult to check in this PR (due to the backward compatibility).

I revert v2,v3 and v4 apis, since they break backward compatibility. Maybe they should be discussed in another PR.

File: tensorflow/python/ops/image_ops_test.py [hunk 1]
```diff
@@ -4956,6 +4957,29 @@ def nms_func(box, score, iou_thres, score_thres):
       scores = constant_op.constant([0.9])
       nms_func(boxes, scores, iou_thres, [[score_thres]])
 
+  @test_util.xla_allow_fallback(
+      "non_max_suppression with dynamic output shape unsupported.")
+  def testTensors(self):
+    with context.eager_mode():
+      boxes_f = open("tensorflow/core/lib/tensors/boxes.tensor", "rb").read()
```
Comment (mingxingtan) [inline tensorflow/python/ops/image_ops_test.py:4964]: This is difficult to read and use. Could you explicitly construct a tensor with a list (or with a numpy random value with fixed seed)?

Comment (fsx950223) [reply tensorflow/python/ops/image_ops_test.py:4964]: > This is difficult to read and use. Could you explicitly construct a tensor with a list (or with a numpy random value with fixed seed)?

Could I remove the test case if it's not necessary?

Comment (fsx950223) [reply tensorflow/python/ops/image_ops_test.py]: > This is difficult to read and use. Could you explicitly construct a tensor with a list (or with a numpy random value with fixed seed)?

Done.

**Comment by mingxingtan** [comment]
Thanks @fsx950223 
The code looks good to me. Running some internal tests now. Will try to get it in asap.

**Review by mingxingtan — APPROVED** [review]

**Pull request merged**